telemetry/definitions/commonDefinitions.json

```diff
@@ -298,19 +298,32 @@
             "description": "The Java version on the user's machine"
         },
         {
-            "name": "codeTransformLocalMavenVersion",
+            "name": "codeTransformLocalBuildSystemVersion",
             "type": "string",
-            "description": "The Maven version on the user's machine"
+            "description": "The build system version on the user's machine"
         },
         {
-            "name": "codeTransformMavenBuildCommand",
+            "name": "codeTransformBuildCommand",
             "type": "string",
-            "description": "Type of maven command",
+            "description": "Type of build command",
             "allowedValues": [
                 "mvnw.cmd",
                 "mvnw",
                 "mvn",
-                "IDEBundledMaven"
+                "IDEBundledMaven",
+                "gradlew.bat",
+                "gradlew",
+                "gradle",
+                "IDEBundledGradle"
+            ]
```

Comment:
For telemetry, is it necessary to enforce a list of allowed values? I think not imposing allowed values will make the metric types more scalable. Same for codeTransformBuildSystem

Reply:
These enum-style metric definitions are purely for developer (client-side) convenience, they aren't enforced by the telemetry service. You can always override them in the client.